### PR TITLE
[receiver/elasticsearch]: add metrics for merge operations with total shard aggregation

### PR DESCRIPTION
**Pull request by aboguszewski-sumo**
**Description:** 
Three new metrics have been added. One was already present in `metadata.yaml`, but was not emitted.

**Link to tracking Issue:** #14635

**Testing:**
Unit and integration

**Documentation:** 
Generated with `mdatagen`.

### CODE REVIEW

**Comment by djaglowski** [comment]
Would you mind splitting off a PR for the one that was mistakenly omitted? This is a bug and should be accepted separately.

The two new metrics seems reasonable to me.

**Comment by aboguszewski-sumo** [comment]
I added the tests.

File: receiver/elasticsearchreceiver/metadata.yaml [hunk 1]
```diff
@@ -758,3 +758,21 @@ metrics:
...
     enabled: true
+  elasticsearch.index.operations.merge.size:
+    description: The total size of merged segments for an index.
+    unit: By
+    sum:
+      monotonic: true
+      aggregation: cumulative
+      value_type: int
+    attributes: [index_aggregation_type]
+    enabled: true
```
Comment (djaglowski) [inline receiver/elasticsearchreceiver/metadata.yaml:769]: Let's make these new metrics disabled by default.

**Comment by aboguszewski-sumo** [comment]
@djaglowski I changed the metrics to disabled by default. There is one thing I've changed in the tests due to that. I decided that it would be the best if these metrics were enabled only in one test (I picked `full` because it seemed the most suitable one). There are two reasons for doing so:
1. This way we test the correctness of the metric, but also make sure that it is disabled by default.
2. Rebases will be much easier to be done. To be honest, the most annoying thing in working on these metrics is modifying the files with test results (and in case of this receiver, there are 5 files in total). If only one test file has to be modified with each PR, the rebases can go more smoothly.
I'd like to follow this convention for other metrics that are going to be disabled by default (I'll be fixing the PRs gradually), so if you have any objection, please raise it now.

**Comment by djaglowski** [comment]
@aboguszewski-sumo, I agree with your reasoning on the test files. This should be a good pattern to follow going forward.

**Review by djaglowski — APPROVED** [review]

**Pull request merged**